crates/tags/src/tag.rs

```diff
@@ -45,3 +46,19 @@ impl FromRequest for Tag {
             .or(Err(StatusCode::BadRequest))
     }
 }
+
+#[cfg(test)]
+mod test {
+    use crate::Tag;
+
+    use std::str::FromStr;
+
+    #[test]
+    fn parse_string() {
+        let test_value = Tag::from_str("1.2.3-beta").unwrap();
```

Comment:
```suggestion
        let test_value: Tag = "1.2.3-beta".parse().unwrap();
```

The reason why this works seems magical at first. There are two traits: `FromStr` and `Parse`. Then there is a blanket implementation of `Parse` so that every implementation of `FromStr` also gets a corresponding implementation of `Parse` automatically. The same thing happens for the traits `From` and `Into` as well as `TryFrom` and `TryInto`. It is a common pattern in Rust.

Reply:
I'm still coming to terms with how Rust hides everything with those mechanisms.